tests/suites/constraints/constraints.sh

```diff
@@ -0,0 +1,60 @@
+run_constraints_aws() {
+  # Echo out to ensure nice output to the test suite.
+  echo
+
+  file="${TEST_DIR}/constraints-aws.txt"
+
+  ensure "constraints-aws" "${file}"
+
+	echo "Deploy 5 machines with different constraints"
+  juju add-machine --constraints "root-disk=16G"
+  juju add-machine --constraints "cores=4"
+  juju add-machine --constraints "cpu-power=30"
+  juju add-machine --constraints "root-disk=16G cpu-power=30"
+  juju add-machine --constraints "instance-type=i3.xlarge"
+
+  wait_for_machine_agent_status "0" "started"
+  wait_for_machine_agent_status "1" "started"
+  wait_for_machine_agent_status "2" "started"
+  wait_for_machine_agent_status "3" "started"
+  wait_for_machine_agent_status "4" "started"
+
+	echo "Ensure machine 0 has 16G root disk"
+  machine0_hardware=$(juju machines --format json | jq -r '.["machines"]["0"]["hardware"]')
+  check_contains "${machine0_hardware}" "root-disk=16384M"
```

Comment:
Officially constraints are "must be at least".
So we shouldn't be doing an exact match on the constraint but a "at least 16GB".
But otherwise this looks good.

Reply:
done. For this purpose, I've added yet another function `check_ge`.